prefect_aws/workers/ecs_worker.py

```diff
@@ -0,0 +1,1343 @@
+        container_overrides = overrides.get("containerOverrides", [])
+
+        # Ensure the network configuration is present if using awsvpc for network mode
+
+        if task_definition.get("networkMode") == "awsvpc" and not task_run_request.get(
+            "networkConfiguration"
+        ):
+            task_run_request["networkConfiguration"] = self._load_vpc_network_config(
+                configuration.vpc_id, boto_session
+            )
+
+        # Ensure the container name is set if not provided at template time
+
+        container_name = (
+            configuration.container_name
+            or _container_name_from_task_definition(task_definition)
+            or ECS_DEFAULT_CONTAINER_NAME
+        )
+
+        if container_overrides and not container_overrides[0].get("name"):
+            container_overrides[0]["name"] = container_name
+
+        # Ensure configuration command is respected post-templating
+
+        container = _get_container(container_overrides, container_name)
+
+        if container:
+            # Override the command if given on the configuration
+            if configuration.command:
+                container["command"] = configuration.command
+
+        # Clean up templated variable formatting
+
+        for container in container_overrides:
+            if isinstance(container.get("command"), str):
+                container["command"] = shlex.split(container["command"])
+            if isinstance(container.get("environment"), dict):
+                container["environment"] = [
+                    {"name": k, "value": v} for k, v in container["environment"].items()
+                ]
+
+        if isinstance(task_run_request.get("tags"), dict):
+            task_run_request["tags"] = [
+                {"key": k, "value": v} for k, v in task_run_request["tags"].items()
+            ]
+
+        if overrides.get("cpu"):
+            overrides["cpu"] = str(overrides["cpu"])
+
+        if overrides.get("memory"):
+            overrides["memory"] = str(overrides["memory"])
+
+        # Ensure configuration tags and env are respected post-templating
+
+        tags = [
+            item
+            for item in task_run_request.get("tags", [])
+            if item["key"] not in configuration.labels.keys()
+        ] + [
+            {"key": k, "value": v}
+            for k, v in configuration.labels.items()
+            if v is not None
+        ]
+        if tags:
+            task_run_request["tags"] = tags
+
+        if container:
+            environment = [
+                item
+                for item in task_run_request.get("environment", [])
+                if item["name"] not in configuration.env.keys()
+            ] + [
+                {"name": k, "value": v}
+                for k, v in configuration.env.items()
+                if v is not None
+            ]
+            if environment:
+                container["environment"] = environment
```

Comment:
This appears to be the most reasonable way to ensure items added to the configuration during `prepare_for_flow_run` are present in the request since preparation occurs _after_ templating they would otherwise be excluded.